### ISSUE TITLE
Podman on Windows has issues with environment variables with spaces in their name

### ISSUE THREAD

**Issue by OOPMan** [containers/podman#15597]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**
It seems that `podman run` on Windows has issues when there is an environment variable set with a space in its name. When this is issue is encountered `podman run` fails completely

**Steps to reproduce the issue:**

1. Install Podman 4.2.0 for Windows

2. `podman machine init`

3. `podman machine start`

4. `set A Test=A`

5. `podman run ubi8-micro date`

**Describe the results you received:**
```C:\Users\Adam Jorgensen>podman run ubi8-micro date
Resolved "ubi8-micro" as an alias (/etc/containers/registries.conf.d/000-shortnames.conf)
Trying to pull registry.access.redhat.com/ubi8-micro:latest...
Getting image source signatures
Checking if image destination supports signatures
Copying blob sha256:d25fc32c2ad84272a6cc207f8fdd1f08f1777e1677b44641fb618ea7d07cd8d9
Copying blob sha256:4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1
Copying config sha256:68f7c3da6edd15545631001312212b754aa09e433c48718fe26b2a6b4c266d38
Writing manifest to image destination
Storing signatures
Error: error parsing host environment variables: name "A Test" has white spaces, poorly formatted name
```

**Describe the results you expected:**
Expected output of the `date` command

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
C:\Users\Adam Jorgensen>podman version
Client:       Podman Engine
Version:      4.2.0
API Version:  4.2.0
Go Version:   go1.16.15
Git Commit:   7fe5a419cfd2880df2028ad3d7fd9378a88a04f4
Built:        Thu Aug 11 10:20:57 2022
OS/Arch:      windows/amd64

Server:       Podman Engine
Version:      4.2.0
API Version:  4.2.0
Go Version:   go1.18.4
Built:        Thu Aug 11 10:42:17 2022
OS/Arch:      linux/amd64
```

**Output of `podman info`:**

```
C:\Users\Adam Jorgensen>podman info
host:
  arch: amd64
  buildahVersion: 1.27.0
  cgroupControllers: []
  cgroupManager: cgroupfs
  cgroupVersion: v1
  conmon:
    package: conmon-2.1.0-2.fc36.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.0, commit: '
  cpuUtilization:
    idlePercent: 99.88
    systemPercent: 0.07
    userPercent: 0.05
  cpus: 16
  distribution:
    distribution: fedora
    variant: container
    version: "36"
  eventLogger: file
  hostname: Killing-Machine
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 5.10.60.1-microsoft-standard-WSL2
  linkmode: dynamic
  logDriver: journald
  memFree: 12983296000
  memTotal: 13321940992
  networkBackend: netavark
  ociRuntime:
    name: crun
    package: crun-1.5-1.fc36.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.5
      commit: 54ebb8ca8bf7e6ddae2eb919f5b82d1d96863dea
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  remoteSocket:
    exists: true
    path: /run/user/1000/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: true
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.0-0.2.beta.0.fc36.x86_64
    version: |-
      slirp4netns version 1.2.0-beta.0
      commit: 477db14a24ff1a3de3a705e51ca2c4c1fe3dda64
      libslirp: 4.6.1
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.3
  swapFree: 4294967296
  swapTotal: 4294967296
  uptime: 0h 6m 44.00s
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  volume:
  - local
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - docker.io
  - quay.io
store:
  configFile: /home/user/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /home/user/.local/share/containers/storage
  graphRootAllocated: 269490393088
  graphRootUsed: 407142400
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 1
  runRoot: /run/user/1000/containers
  volumePath: /home/user/.local/share/containers/storage/volumes
version:
  APIVersion: 4.2.0
  Built: 1660228937
  BuiltTime: Thu Aug 11 10:42:17 2022
  GitCommit: ""
  GoVersion: go1.18.4
  Os: linux
  OsArch: linux/amd64
  Version: 4.2.0
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

N/A. Installed using https://github.com/containers/podman/releases/download/v4.2.0/podman-v4.2.0.msi

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/main/troubleshooting.md)**

Yes

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Tested on two different physical system running 21H1

**Comment by mheon** [comment]
@rhatdan You fixed this, right?

**Comment by edsantiago** [comment]
Fixed in #15434, which will presumably make it out in an upcoming podman release.

**Comment by OOPMan** [comment]
Thanks!
